bareops-main/src/main.rs

```diff
@@ -1,12 +1,131 @@
+        &mut self.table
+    }
+    fn ctx(&mut self) -> &mut WasiCtx {
+        &mut self.ctx
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), BareopsError> {
+    let cmd = clap::Command::new("bareops")
+        .subcommand_required(true)
+        .subcommand(
+            clap::Command::new("check").arg(
+                clap::Arg::new("file")
+                    .required(true)
+                    .value_parser(clap::value_parser!(std::path::PathBuf))
+                    .short('f')
+                    .help("file to run"),
+            ),
+        )
+        .subcommand(
+            clap::Command::new("run").arg(
+                clap::Arg::new("file")
+                    .required(true)
+                    .value_parser(clap::value_parser!(std::path::PathBuf))
+                    .short('f')
+                    .help("file to run"),
+            ),
+        );
 
-fn main() {
-    let source = NamedSource::new("demo", "...".to_string());
-    let _tasks = match bareops_lang::parse(&source) {
-        Err(e) => {
-            eprintln!("{:?}", e);
-            return;
+    let matches = cmd.get_matches();
+    let path = match matches.subcommand() {
+        Some(("check", matches)) => {
+            let file = matches.get_one::<PathBuf>("file");
+
+            // TODO: parse tasks file instead of demo
+            let source = NamedSource::new("demo", "...".to_string());
+            let _tasks = match bareops_lang::parse(&source) {
+                Err(e) => {
+                    eprintln!("{:?}", e);
+                    return Ok(());
+                }
+                Ok(tasks) => tasks,
+            };
+            file
+        }
+        Some(("run", matches)) => {
+            // TODO: parse tasks file
+            let file = matches.get_one::<PathBuf>("file");
+            file
+        }
+        _ => {
+            // should not happen, if clap parses correctly
+            return Err(BareopsError::ParseFailure("Wrong usage".to_string()));
         }
-        Ok(tasks) => tasks,
-    };
+    }
+    .ok_or(BareopsError::ParseFailure("Wrong usage".to_string()))?;
+
+    // TODO run tasks instead of plugin (for debug now)
+    // TODO load plugins via configured plugin paths
+    let _result = run_plugin(path).await?;
+    Ok(())
+}
+
+async fn run_plugin(path: &PathBuf) -> Result<[Val; 1], BareopsError> {
+    let mut config = Config::new();
+    config.async_support(true);
+    config.wasm_component_model(true);
+    config.debug_info(true);
+
+    let engine =
+        Engine::new(&config).map_err(|e| BareopsError::TaskbookExecutionFailure(e.to_string()))?;
+
+    // bindgen!({world: "plugin", path: "../bareops-plugin-api/wit/world.wit", async: true});
+
+    let mut linker = Linker::<BareopsState>::new(&engine);
+    wasmtime_wasi::add_to_linker_async(&mut linker)
+        .map_err(|e| BareopsError::TaskbookExecutionFailure(e.to_string()))?;
+
+    let component = Component::from_file(&engine, path)
```

Comment:
I think we don't need to re-compile our wasm bytecode into native code each time a plugin is called